packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts

```diff
@@ -207,21 +222,110 @@ export const useGridCellSelection = (
       apiRef.current.rootElementRef?.current?.classList.add(
         gridClasses['root--disableUserSelection'],
       );
+
+      const document = ownerDocument(apiRef.current.rootElementRef?.current);
+      document.addEventListener('mouseup', handleMouseUp, { once: true });
     },
-    [apiRef, hasClickedValidCellForRangeSelection],
+    [apiRef, handleMouseUp, hasClickedValidCellForRangeSelection],
   );
 
-  const handleCellMouseUp = React.useCallback<GridEventListener<'cellMouseUp'>>(() => {
-    lastMouseDownCell.current = null;
-    apiRef.current.rootElementRef?.current?.classList.remove(
-      gridClasses['root--disableUserSelection'],
-    );
-  }, [apiRef]);
+  const handleMouseOver = useEventCallback((event: MouseEvent) => {
+    const virtualScrollerRect = apiRef.current.virtualScrollerRef?.current?.getBoundingClientRect();
+
+    if (!virtualScrollerRect) {
+      return;
+    }
+
+    const isInsideVirtualScroller =
+      event.clientX <= virtualScrollerRect.right &&
+      event.clientX >= virtualScrollerRect.left &&
+      event.clientY >= virtualScrollerRect.top &&
+      event.clientY <= virtualScrollerRect.bottom;
+
+    if (!isInsideVirtualScroller) {
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      stopAutoScroll();
+    }
+  });
+
+  const stopAutoScroll = React.useCallback(() => {
+    if (autoScrollInterval.current) {
+      clearTimeout(autoScrollInterval.current);
+      autoScrollInterval.current = null;
+    }
+
+    if (apiRef.current.virtualScrollerRef?.current) {
+      const document = ownerDocument(apiRef.current.virtualScrollerRef.current);
+      document.removeEventListener('mouseover', handleMouseOver);
+    }
+  }, [apiRef, handleMouseOver]);
 
   const handleCellFocusIn = React.useCallback<GridEventListener<'cellFocusIn'>>((params) => {
     cellWithVirtualFocus.current = { id: params.id, field: params.field };
   }, []);
 
+  const startAutoScroll = React.useCallback(() => {
+    if (autoScrollInterval.current) {
+      return;
+    }
+
+    if (!apiRef.current.virtualScrollerRef?.current) {
+      return;
+    }
+
+    const document = ownerDocument(apiRef.current.virtualScrollerRef.current);
+    document.addEventListener('mouseover', handleMouseOver);
+
+    autoScrollInterval.current = setInterval(() => {
+      if (!apiRef.current.virtualScrollerRef?.current) {
+        return;
+      }
+
+      const virtualScrollerRect =
+        apiRef.current.virtualScrollerRef?.current?.getBoundingClientRect();
```

Comment:
Did you consider moving the `getBoundingClientRect` call outside of `setInterval`?
[`getBoundingClientRect` triggers forced reflow / layout](https://gist.github.com/paulirish/5d52fb081b3570c81e3a), so maybe we can avoid calling it too often assuming that virtual scroller size is constant during the scrolling?